### PR TITLE
Release v0.3.0

### PR DESCRIPTION
**Pull request by mbjohns**
## Release: EXAUQ-Toolbox v0.3.0

This minor release adds support for multi-level adaptive sampling and improves integration with simulation logs. It remains backward-compatible and includes extended test coverage.

### Highlights

- New functionality for preparing multi-level training data:
  - `create_data_for_multi_level_loo_sampling(...)`  
  - `compute_delta_coefficients(...)`
  - `_remove_multi_level_repeated_input(...)`

- Simulation log improvements:
  - `get_simulations()` now returns level information
  - New `prepare_training_data()` method to extract `MultiLevel[TrainingDatum]` from the log

- Extensive unit tests added for all new functionality, covering edge cases and warning behaviour

- Minor cleanup and version bump: `0.2.1` → `0.3.0`

### CODE REVIEW

File: exauq/core/designers.py [hunk 1]
```diff
@@ -1375,3 +1376,219 @@ def compute_multi_level_loo_samples(
...
+
+    if isinstance(correlations, Real):
+        correlations = MultiLevel(
+            {level: correlations for level in levels if level < max(levels)}
+        )
+
+    delta_coefficients = MultiLevel(
+        {max(levels): 1}
+        | correlations.map(
+            lambda k, _: math.prod(
```
[the remark this answers is not in the document]
Comment (mbjohns) [reply exauq/core/designers.py:1547]: The math module IS imported (ln 63)....

File: exauq/core/designers.py [hunk 2]
```diff
@@ -1375,3 +1376,219 @@ def compute_multi_level_loo_samples(
...
+    data:
+        Training data to be checked for repeated input
+    datum:
+        Single TrainingDatum used for comparison of input.
+    level:
+        The highest fidelity level at which the input appears.
+
+    Returns
+    -------
+    MultiLevel[Sequecnce[TrainingDatum]]
```
[the remark this answers is not in the document]
Comment (mbjohns) [reply exauq/core/designers.py:1580]: Good spot AI!

File: exauq/core/designers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JSalter90) [reply exauq/core/designers.py:1547]: 'mistakes are possible', indeed

**Review by JSalter90 — APPROVED** [review]
All tests run locally 🚀
